hubblestack/extmods/module_runner/runner.py

```diff
@@ -79,32 +79,57 @@ def init_loader(self):
                                     pack={'__salt__': __salt__,
                                         '__grains__': __grains__})
 
-        ## Initializing comparator only for Audit
-        if self._caller == Caller.AUDIT:
-            global __comparator__
-            __comparator__ = salt.loader.LazyLoader(salt.loader._module_dirs(__opts__, 'comparators'),
-                                        __opts__,
-                                        tag='comparators',
-                                        pack={'__salt__': __salt__,
-                                            '__grains__': __grains__})
-            hubblestack.extmods.module_runner.comparator.__comparator__ = __comparator__
+        # Comparator can be needed in both Audit/FDG
+        global __comparator__
+        __comparator__ = salt.loader.LazyLoader(salt.loader._module_dirs(__opts__, 'comparators'),
+                                    __opts__,
+                                    tag='comparators',
+                                    pack={'__salt__': __salt__,
+                                        '__grains__': __grains__})
+        hubblestack.extmods.module_runner.comparator.__comparator__ = __comparator__
 
 
     ######################################################
     ################# Non-Public methods #################
     ######################################################
 
-    def _validate_module_params(self, module_name, profile_id, module_args):
+    def _validate_module_params(self, module_name, profile_id, module_args, extra_args=None):
         """
         A helper method to invoke module's validate_params method
         """
+        if not module_args:
+            raise CommandExecutionError('Could not execute block \'{0}\', as it is not found.'
+                                        .format(profile_id))
+
         validate_param_method = '{0}.validate_params'.format(module_name)
-        __hmods__[validate_param_method](profile_id, module_args)
+        if extra_args:
```

Comment:
Instead of using if else here, we should simply pass extra_args to validate param method and handle the scenario there if extra_args is not None.